lib/core-common/src/types.ts

```diff
@@ -439,3 +422,14 @@ export interface StorybookConfig {
    */
   storyIndexers?: (indexers: StoryIndexer[], options: Options) => StoryIndexer[];
 }
+
+export type PresetProperty<K, TStorybookConfig = StorybookConfig> =
```

Comment:
what's the idea here?

Reply:
Great question!

So I use it in another file. The idea is that presets can expose a value like `addons`
This value can be written as a direct value `[]`. But it can also be a function (async or sync) that returns the value.

We've hard-coded that some properties can only be a function, such as `webpackFinal`.

This type is a generic that can be used like this:
```
PresetProperty<'core', StorybookConfig>
```
... where providing `StorybookConfig` is optional, if will fallback to the core-common `StorybookConfig` interface.

This allows for a lot of flexibility, but also type-safety / strictness.